crates/rome_js_analyze/src/analyzers/nursery/no_misused_new.rs

```diff
@@ -0,0 +1,232 @@
+                "Type alias can't use `constructor`."
+            })
+            .to_owned(),
+        }
+    }
+
+    fn range(&self) -> TextRange {
+        match self {
+            RuleState::ClassMisusedNew(range)
+            | RuleState::InterfaceMisusedNew(range)
+            | RuleState::InterfaceMisusedConstructor(range)
+            | RuleState::TypeAliasMisusedConstructor(range) => *range,
+        }
+    }
+}
+
+impl Rule for NoMisusedNew {
+    type Query = Ast<DeclarationQuery>;
+    type State = RuleState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        match node {
+            DeclarationQuery::TsInterfaceDeclaration(decl) => {
+                let interface_ident = decl.id().ok()?.name_token().ok()?;
+                for member in decl.members() {
+                    match member {
+                        AnyTsTypeMember::TsConstructSignatureTypeMember(construct) => {
+                            if construct.new_token().ok().is_none() {
+                                continue;
+                            }
+                            let any_ts_type = construct.type_annotation()?.ty().ok()?;
+                            let reference_type = any_ts_type.as_ts_reference_type()?;
+                            let return_type_ident = extract_return_type_ident(&reference_type)?;
+                            if interface_ident.text_trimmed() == return_type_ident.text_trimmed() {
+                                return Some(RuleState::InterfaceMisusedNew(construct.range()));
+                            }
+                        }
+                        AnyTsTypeMember::TsMethodSignatureTypeMember(method) => {
+                            let method_name = method.name().ok()?;
+                            let method_name = method_name.as_js_literal_member_name()?.syntax();
+                            if method_name.text_trimmed() == "constructor" {
+                                return Some(RuleState::InterfaceMisusedConstructor(
+                                    method.range(),
+                                ));
+                            }
+                        }
+                        _ => continue,
+                    };
+                }
+                None
+            }
+            DeclarationQuery::TsTypeAliasDeclaration(decl) => {
+                let any_ts_type = decl.ty().ok()?;
+                let object = any_ts_type.as_ts_object_type()?;
+                let method = object
+                    .members()
+                    .into_iter()
+                    .filter_map(|member| member.as_ts_method_signature_type_member().cloned())
+                    .next()?;
+                let method_name = method.name().ok()?;
+                let method_name = method_name.as_js_literal_member_name()?.syntax();
+                if method_name.text_trimmed() == "constructor" {
```

Comment:
I think we can simplify to:

```suggestion
                if method_name.name()?.text() == "constructor" {
```

This has also the benefit of handling computed key like `["constructor"]`.